### PR TITLE
fix(datastore-v1): correct the filter predicate logic applied to optional fields

### PR DESCRIPTION
**Pull request by 5d**
## Description

The GraphQL filter predication not working properly on optional field. It always return `false` on optional field with value `nil`.

## General Checklist

- [ ] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [X] PR title conforms to conventional commit style
- [ ] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderConfirmSignupTests.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ class AuthenticationProviderConfirmSignupTests: BaseAuthenticationProviderTest {
     ///
     func testSuccessfulConfirmSignUp() {
 
-        let mockSignupResult = SignUpResult(signUpState: .confirmed, codeDeliveryDetails: nil)
+        let mockSignupResult = SignUpResult(signUpState: .confirmed, codeDeliveryDetails: nil, userSub: nil)
```
Comment (lawmicha) [inline AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderConfirmSignupTests.swift:29]: what's this from?

File: AmplifyPlugins/Core/AWSPluginsCoreTests/Query/QueryPredicateEvaluateGeneratedBoolTests.swift [hunk 2]
```diff
@@ -41,7 +41,7 @@ class QueryPredicateEvaluateGeneratedBoolTests: XCTestCase {
 
         let evaluation = try predicate.evaluate(target: instance.eraseToAnyModel().instance)
 
-        XCTAssertFalse(evaluation)
+        XCTAssertTrue(evaluation)
```
Comment (lawmicha) [inline AmplifyPlugins/Core/AWSPluginsCoreTests/Query/QueryPredicateEvaluateGeneratedBoolTests.swift:44]: [nit] let's add either a test comment or improve the test name to `testBooltruenotEqualBoolNil` it's not immediately obvious that the `instance` has a nil Bool due to the default parameter value. Here and other tests that have similar changes

File: AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderConfirmSignupTests.swift (hunk 1 above)
Comment (5d) [reply AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderConfirmSignupTests.swift:29]: I think this is from the modle update of AWSCognito
